### PR TITLE
IGNITE-23443 Create Gradle task to update versions

### PR DESCRIPTION
**Pull request by valepakh**
https://issues.apache.org/jira/browse/IGNITE-23443

This adds a new project property `projectVersion` and a new task `:platforms:updateVersion`.
The property must be present either in the `gradle.properties` file or in the command line.
It determines the version of the Java projects.
C/C++ CMake projects, .NET and Python projects versions needs to be updated using the new task.

To update the version, first set it in the `gradle.properties` file, then run `./gradlew :platforms:updateVersion`.
After that commit updated files.

### CODE REVIEW

File: modules/platforms/build.gradle [hunk 1]
```diff
@@ -206,10 +209,46 @@ artifacts {
             odbc(file("$buildDir/cpp/lib/libignite3-odbc.so")) {
                 builtBy cmakeBuildOdbc
             }
-            //TODO https://issues.apache.org/jira/browse/IGNITE-23443
-            odbc(file("$buildDir/cpp/lib/libignite3-odbc.so.3.0.0")) {
+            odbc(file("$buildDir/cpp/lib/libignite3-odbc.so.$projectVersion")) {
```
Comment (isapego) [inline modules/platforms/build.gradle:212]:
```suggestion
            odbc(file("$buildDir/cpp/lib/libignite3-odbc.so.$cmakeProjectVersion(version)")) {
```

File: modules/platforms/build.gradle [hunk 2]
```diff
@@ -190,8 +192,9 @@ artifacts {
         cppClientHeaders(file("$buildDir/install/include")) {
             builtBy cmakeBuildClient
         }
-        //TODO https://issues.apache.org/jira/browse/IGNITE-23443
-        cppClient(file("$buildDir/cpp/lib/libignite3-client.so.3.0.0")) {
+
+        def projectVersion = cmakeProjectVersion(project.projectVersion.toString())
+        cppClient(file("$buildDir/cpp/lib/libignite3-client.so.$projectVersion")) {
```
[the remark this answers is not in the document]
Comment (isapego) [reply modules/platforms/build.gradle:197]: I know there should be a variable, it's just to give you an idea of what should be fixed here.

File: modules/platforms/cpp/CMakeLists.txt [hunk 3]
```diff
@@ -16,7 +16,14 @@
 #
 
 cmake_minimum_required(VERSION 3.18)
-project(Ignite.C++ VERSION 3.0.0 LANGUAGES CXX)
+
+file(READ _version.txt IGNITE_VERSION)
+if(NOT DEFINED IGNITE_VERSION)
+  set(IGNITE_VERSION 3.0.0)
+  message(WARNING "Failed to read version from File. Using default version \"${IGNITE_VERSION}\".")
```
Comment (isapego) [inline modules/platforms/cpp/CMakeLists.txt:23]:
```suggestion
  message(WARNING "Failed to read version from _version.txt file. Using default version \"${IGNITE_VERSION}\".")
```

File: modules/platforms/python/setup.py [hunk 4]
```diff
@@ -46,12 +47,19 @@ def is_a_requirement(req_line):
     long_description = readme_file.read()
 
 version = None
-with open(PACKAGE_NAME + '/__init__.py', 'r') as fd:
-    version = re.search(r'^__version__\s*=\s*[\'"]([^\'"]*)[\'"]',
-                        fd.read(), re.MULTILINE).group(1)
+with open(PACKAGE_NAME + '/_version.txt', 'r') as fd:
```
Comment (isapego) [inline modules/platforms/python/setup.py:50]:
```suggestion
with open(os.path.join(PACKAGE_NAME, '_version.txt'), 'r') as fd:
```

**Review by isapego — CHANGES_REQUESTED** [review]
See inline

**Review by isapego — APPROVED** [review]

**Review by ptupitsyn — APPROVED** [review]

**Pull request merged**